pkg/depository/create.go

```diff
@@ -17,18 +17,241 @@ limitations under the License.
 package depository
 
 import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"github.com/bestchains/bc-cli/pkg/common"
+	uhttp "github.com/bestchains/bc-cli/pkg/utils/http"
+	"github.com/bestchains/bestchains-contracts/library/context"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
 
+	"github.com/bestchains/bestchains-contracts/library"
 	"github.com/spf13/cobra"
 )
 
 func NewCreateDepositoryCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use: "depository NAME [args]",
-		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Println("create depository")
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// Get depot info from flags
+			n, err := cmd.Flags().GetString("name")
+			if err != nil {
+				return err
+			}
+			t, err := cmd.Flags().GetString("contentType")
+			if err != nil {
+				return err
+			}
+			id, err := cmd.Flags().GetString("contentID")
+			if err != nil {
+				return err
+			}
+			p, err := cmd.Flags().GetString("platform")
+			if err != nil {
+				return err
+			}
+			acc, err := cmd.Flags().GetString("wallet-account")
+			if err != nil {
+				return err
+			}
+			key, err := cmd.Flags().GetString("privateKey")
+			if err != nil {
+				return err
+			}
+			accGen, err := cmd.Flags().GetBool("wallet-account-generate")
+			if err != nil {
+				return err
+			}
+			untrusted, err := cmd.Flags().GetBool("untrusted")
+			if err != nil {
+				return err
+			}
+
+			// FIXME: the host should be read from the configuration file.
+			host, _ := cmd.Flags().GetString("host")
+			if host == "" {
+				return fmt.Errorf("no host provided")
+			}
+
+			valueBase64 := generateValueDepotBase64(n, t, id, p)
+
+			if !accGen {
+				// FIXME: key auth to be done, bypassed temporally
+				// key auth
+				var privKey ecdsa.PrivateKey
+				err = json.Unmarshal([]byte(key), &privKey)
+				if err != nil {
```

Comment:
1. Is the `key` a filename or a private key string?
2. Is it correct to use json to parse the ecdsa private key?

```go
    a, _ := pem.Decode([]byte(x))
    pk, err := x509.ParseECPrivateKey(a.Bytes)
    if  err != nil {
        log.Fatal("ParseECPrivateKey error ", err)
    }
```

Reply:
I assumed `key` is a password-like `string` before so I handled it with `json`. Should the `privateKey` be saved into & parsed from local file?

Reply:
@Lanture1064 @0xff-dev as we discussed, we manage key paris with another command `bc-cli  create/get/delete account ` in https://github.com/bestchains/bc-cli/issues/8